### PR TITLE
feat(TDI-40535): track application in snowflake

### PR DESCRIPTION
**Pull request by olhava**
**What is the current behavior?** (You can also link to an open issue here)
No info about application in connection string

**What is the new behavior?**
Added application parameter to connection string

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java [hunk 1]
```diff
@@ -309,6 +312,14 @@ public String getConnectionUrl() {
         return url.toString();
     }
 
+    private String getApplication() {
+        String application = "Talend-";
+        if (!StringUtils.isEmpty(talendProductVersion)) {
```
Comment (mbasiuk-talend) [inline components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/SnowflakeConnectionProperties.java]: 1. If you use StringUtils, you might refer to method isNotEmpty it'll be more informative, as for me.
2. What if talendProductVersion is null, you will set appName "Talend-" ? Is there any better name if version is missing?

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/60/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/60/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aovarnavina%2FTDI-40535_track_application_in_snowflake)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/69/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/69/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aovarnavina%2FTDI-40535_track_application_in_snowflake)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/72/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/72/)

**Pull request merged**